pacman/utilities/utility_objs/resource_tracker.py

```diff
@@ -147,6 +156,44 @@ def __init__(self, machine, chips=None):
             for x, y in chips:
                 self._chips_available.add((x, y))
 
+    def _convert_pre_allocated_resources(self, pre_allocated_resources):
+        """ allocates pre allocated sdram and specific cores to the trackers.
+        also builds an arbitrary core map for usage throughout resource tracker
+
+        :param pre_allocated_resources: the pre allocated resources from the\
+         tools
+        :type pre_allocated_resources: PreAllocatedResourceContainer
+        :return: a mapping of chip to arbitrary core demands
+        :rtype: dict of key (int, int) to int
+        """
+
+        # remove sdram by adding the cost into the sdram tracker already
+        if pre_allocated_resources is not None:
```

Comment:
I would prefer to invert the test and structure the case for `pre_allocated_resource is None` to look like an early exit due to a precondition not being satisfied, allowing the rest to be de-indented.